### PR TITLE
Fix world age problems when running tests

### PR DESCRIPTION
**Pull request by timholy**
This WiP solves the world age problems by an inversion of control: rather than having `lower_incrementally(f, mod, expr)` recurse into `expr`, lower its individual pieces, and then running the resulting frames with `f`, this redesigns `prepare_toplevel` so that it splits `expr` into frames and returns it as a task list. The caller is then responsible for running each of them. While this is extra work for the caller, the big advantage is that if you do this in top-level you can ensure that there is a world age update after each new method definition. (This is achieved just by returning "prematurely" after each 3-arg `:method` expr is encountered.)

The core tests pass, but I'm having surprising difficulty rewriting `juliatests.jl` to use this. I may have to pivot to other things for the remainder of the day, so please don't merge yet. (I did want people to know this is coming---it's a pretty disruptive change.)

Bonus: includes new docs!

### CODE REVIEW

**Comment by KristofferC** [comment]
Is is true that theoretically, we could just `invokelatest` every function? In that case, perhaps it should be mentioned why we don't do that.

**Comment by timholy** [comment]
That was indeed the old bandaid, you can find such a call in `lower!`. But that became not-very-viable when I discovered that I'd also need it on calls in `maybe_evaluate_builtin`: it would make the interpreter insanely slow. Since this is an issue that affects the test suite much more than ordinary usage via `@interpret foo(args...)`, I thought it was better to isolate the pain.

**Comment by KristofferC** [comment]
Yeah, I didn't mean to suggest it as a workaround but just having a note about it in the docs to immediately catch the people going "oh they don't know about `invokelatest` to get around world error"-people.

**Comment by timholy** [comment]
Makes total sense, great idea.

**Comment by pfitzseb** [comment]
Maybe I'm missing something, but wouldn't it be possible to insert a manual world age increment after `:method` expressions?

**Comment by timholy** [comment]
I don't know how to do that, do you?

**Comment by pfitzseb** [comment]
> I don't know how to do that, do you?

I don't -- turns out I didn't actually understand the problem properly (I thought the global world age wasn't incremented properly, but the problem is that the "local" world age is constant despite a new method being defined). So yeah, I can't think of a better way to do this.

**Comment by timholy** [comment]
@pfitzseb, I have long suspected that the omission of manual world-age updates is deliberate, but I hadn't really thought carefully until prompted by your question. Now my guess is this: if Julia supported manual updates, what would happen for a compiled function that happened to redefine itself and then trigger a world age update? Or a function that redefined one of its callers?

In contrast, if world age only updates when you are at top level, there is no caller (well, there's potentially `eval_user_input`, but this is a special case thats easily caught). So that situation can never arise.

Note that base "cheats" a bit on this: see, for example, https://github.com/JuliaLang/julia/blob/68db87147f9806db605c4df9f32b0869554cf4b7/src/interpreter.c#L883-L885. Since we can't do that, we have to employ other strategies.

**Comment by timholy** [comment]
I now have more of an idea of why I'm struggling with this. Check out this bizarre Julia bug:
```julia
...
    frame.pc[] = pc
    # Handle the return
    stmt = pc_expr(frame, pc)
    if nstmts == 0 && !isexpr(stmt, :return)
        println("aborted")
        JuliaInterpreter.show_stackloc(stack, frame, pc)
        @show pc frame
        ret = Aborted(frame, pc)
        @show ret
        return ret, nstmts
    end
...
```
and that `Aborted` constructor looks like this
```julia
function Aborted(frame::JuliaStackFrame, pc)
    println("Aborted constructor")
    @show pc frame
    lineidx = frame.code.code.codelocs[convert(Int, pc)]
    @show lineidx
    return Aborted(frame.code.code.linetable[lineidx])
end
```
yet I'm seeing output like
```
aborted
Main.JuliaTests
  Main.JuliaTests
    showerror(io::IO, ex::MethodError) in Base at errorshow.jl:168
      showerror_ambiguous(io::IO, meth, f, args) in Base at errorshow.jl:271
        print(io::IO, xs...) in Base at strings/io.jl:39
          print(io::IO, x) in Base at strings/io.jl:29
            show(io::IO, m::Method) in Base at methodshow.jl:113
              #show#520(kwtype::Union{Nothing, DataType}, ::Any, io::IO, m::Method) in Base at methodshow.jl:113
                arg_decl_parts(m::Method) in Base at methodshow.jl:55
                  argtype_decl(env, n, sig::DataType, i::Int64, nargs, isva::Bool) in Base at methodshow.jl:6
                    string_with_env(env, xs...) in Base at strings/io.jl:129
                      print(io::IO, x) in Base at strings/io.jl:29
                        show(io::IO, x::DataType) in Base at show.jl:436
                          show_datatype(io::IO, x::DataType) in Base at show.jl:511
                            show_type_name(io::IO, tn::Core.TypeName) in Base at show.jl:450
                              print(io::IO, s::Union{SubString{String}, String}) in Base at strings/io.jl:167
                                write(io::IO, s::Union{SubString{String}, String}) in Base at strings/io.jl:165
                                  unsafe_write(io::Base.AbstractPipe, p::Ptr{UInt8}, nb::UInt64) in Base at io.jl:225
                                    unsafe_write(to::Base.GenericIOBuffer, p::Ptr{UInt8}, nb::UInt64) in Base at iobuffer.jl:409
                                      getproperty(x, f::Symbol) in Base at sysimg.jl:18, pc = 2
pc = JuliaProgramCounter(2)
frame = JuliaStackFrame(JuliaInterpreter.JuliaFrameCode(getproperty(x, f::Symbol) in Base at sysimg.jl:18, CodeInfo(
1 ─      nothing
│   %2 = ($(QuoteNode(getfield)))(x, f)
└──      return %2
), #= stuff =#)
Aborted constructor
pc = JuliaProgramCounter(140359322445712)   # what?!? wasn't it at 2 moments ago?
frame = JuliaStackFrame(JuliaInterpreter.JuliaFrameCode(getproperty(x, f::Symbol) in Base at sysimg.jl:18, CodeInfo(
1 ─      nothing
│   %2 = ($(QuoteNode(getfield)))(x, f)
└──      return %2
...
```

**Comment by KristofferC** [comment]
Yikes 😨

**Comment by timholy** [comment]
OK, this seems to work now, and adds in a fix for generated functions as a bonus. With this the tuple test finishes.

*However*, one negative is that this disables the parallel framework and runs all tests in `id 1`. Because of the tests that kill Julia, this means we don't finish them. I haven't yet looked into how to run these tests in top level on the workers (all the `@async` stuff goes through thunks, AFAIU). Maybe I need to `eval` them on the workers? Thoughts?

I'll point out another oddity in the code.

File: test/toplevel.jl [hunk 1]
```diff
@@ -140,8 +147,8 @@ module Toplevel end
     @test @interpret(Toplevel.paramtype(Vector)) == Toplevel.NoParam
     @test @interpret(Toplevel.Inner.g()) == 5
     @test @interpret(Toplevel.Inner.InnerInner.g()) == 6
-    @test @interpret(isdefined(Toplevel, :Beat))
-    @test @interpret(Toplevel.Beat <: Toplevel.DatesMod.Period)
+    # @test @interpret(isdefined(Toplevel, :Beat))
+    # @test @interpret(Toplevel.Beat <: Toplevel.DatesMod.Period)
```
Comment (timholy) [inline test/toplevel.jl]: Bizarrely, these seem to break the printing of the TestSet results. I haven't looked into the cause.

**Comment by timholy** [comment]
Dang, passed on a 2-day old master. I'll update and see what might need fixing.

**Comment by timholy** [comment]
Here's an update on the tests with this branch (I had to comment out the `@sync` wrapper and put in a `sleep(300)` to let still-running tests finish). It's not an unequivocal win, there are even some regressions. More work to do, I fear :cry:.

Julia Version 1.2.0-DEV.344
Commit a44b421138* (2019-02-16 16:52 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, haswell)
Environment:
  JULIA_CPU_THREADS = 8
Test run at: 2019-02-16T17:24:21.508

Maximum number of statements per lowered expression: 1000000

| Test file | Passes | Fails | Errors | Broken | Aborted blocks |
| --------- | ------:| -----:| ------:| ------:| --------------:|
| ambiguous | 62 | 0 | 0 | 2 | 0 |
| subarray | 278 | 0 | 1 | 0 | 1 |
| strings/basic | 7211 | 1 | 4 | 0 | 3 |
| strings/search | 522 | 0 | 0 | 0 | 0 |
| strings/util | 449 | 0 | 0 | 0 | 0 |
| strings/io | 40 | 0 | 0 | 0 | 1 |
| strings/types | X | X | X | X | X |
| unicode/utf8 | 19 | 0 | 0 | 0 | 0 |
| core | X | X | X | X | X |
| keywordargs | X | X | X | X | X |
| numbers | X | X | X | X | X |
| subtype | X | X | X | X | X |
| char | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| triplequote | 28 | 0 | 0 | 0 | 0 |
| intrinsics | 44 | 0 | 0 | 0 | 0 |
| dict | X | X | X | X | X |
| hashing | X | X | X | X | X |
| iobuffer | 151 | 0 | 0 | 0 | 1 |
| staged | X | X | X | X | X |
| offsetarray | X | X | X | X | X |
| arrayops | X | X | X | X | X |
| tuple | 488 | 0 | 1 | 0 | 0 |
| reduce | 1413 | 0 | 3 | 0 | 4 |
| reducedim | X | X | X | X | X |
| abstractarray | X | X | X | X | X |
| intfuncs | 4364 | 0 | 15 | 0 | 1 |
| simdloop | 61 | 3 | 0 | 0 | 1 |
| vecelement | X | X | X | X | X |
| rational | X | X | X | X | X |
| bitarray | X | X | X | X | X |
| copy | 493 | 11 | 7 | 0 | 0 |
| math | X | X | X | X | X |
| fastmath | 907 | 3 | 3 | 0 | 0 |
| functional | X | X | X | X | X |
| iterators | 601 | 0 | 4 | 0 | 2 |
| operators | 3758 | 0 | 0 | 0 | 1 |
| path | X | X | X | X | X |
| ccall | X | X | X | X | X |
| parse | X | X | X | X | X |
| loading | X | X | X | X | X |
| bigint | X | X | X | X | X |
| sorting | 985 | 0 | 65 | 0 | 4 |
| spawn | X | X | X | X | X |
| backtrace | X | X | X | X | X |
| exceptions | 28 | 19 | 6 | 0 | 0 |
| file | X | X | X | X | X |
| read | X | X | X | X | X |
| version | 1134 | 0 | 0 | 0 | 1 |
| namedtuple | 161 | 0 | 8 | 1 | 0 |
| mpfr | 758 | 0 | 45 | 0 | 0 |
| broadcast | X | X | X | X | X |
| complex | X | X | X | X | X |
| floatapprox | 49 | 0 | 0 | 0 | 0 |
| reflection | X | X | X | X | X |
| regex | 38 | 0 | 1 | 0 | 0 |
| float16 | 124 | 0 | 0 | 0 | 0 |
| combinatorics | 92 | 0 | 3 | 0 | 1 |
| sysinfo | 2 | 0 | 0 | 0 | 0 |
| env | X | X | X | X | X |
| rounding | X | X | X | X | X |
| ranges | X | X | X | X | X |
| mod2pi | 80 | 0 | 0 | 0 | 0 |
| euler | 4 | 0 | 4 | 0 | 4 |
| show | X | X | X | X | X |
| client | 0 | 3 | 0 | 0 | 0 |
| errorshow | X | X | X | X | X |
| sets | X | X | X | X | X |
| goto | X | X | X | X | X |
| llvmcall | X | X | X | X | X |
| llvmcall2 | 6 | 0 | 0 | 0 | 0 |
| grisu | 683 | 0 | 0 | 0 | 1 |
| some | 64 | 0 | 0 | 0 | 0 |
| meta | X | X | X | X | X |
| stacktraces | X | X | X | X | X |
| docs | X | X | X | X | X |
| misc | X | X | X | X | X |
| threads | X | X | X | X | X |
| enums | X | X | X | X | X |
| cmdlineargs | X | X | X | X | X |
| int | X | X | X | X | X |
| checked | X | X | X | X | X |
| bitset | 192 | 0 | 0 | 0 | 0 |
| floatfuncs | 68 | 0 | 1 | 0 | 1 |
| boundscheck | X | X | X | X | X |
| error | 31 | 0 | 0 | 0 | 0 |
| cartesian | 7 | 0 | 0 | 0 | 0 |
| osutils | 42 | 0 | 0 | 0 | 0 |
| channels | X | X | X | X | X |
| iostream | 11 | 0 | 2 | 0 | 0 |
| secretbuffer | 17 | 0 | 0 | 0 | 0 |
| specificity | X | X | X | X | X |
| reinterpretarray | 91 | 0 | 0 | 0 | 1 |
| syntax | X | X | X | X | X |
| logging | X | X | X | X | X |
| missing | X | X | X | X | X |
| asyncmap | X | X | X | X | X |
| SparseArrays/higherorderfns | X | X | X | X | X |
| SparseArrays/sparse | X | X | X | X | X |
| SparseArrays/sparsevector | X | X | X | X | X |
| Pkg/resolve | X | X | X | X | X |
| LinearAlgebra/triangular | X | X | X | X | X |
| LinearAlgebra/qr | X | X | X | X | X |
| LinearAlgebra/dense | X | X | X | X | X |
| LinearAlgebra/matmul | X | X | X | X | X |
| LinearAlgebra/schur | X | X | X | X | X |
| LinearAlgebra/special | X | X | X | X | X |
| LinearAlgebra/eigen | X | X | X | X | X |
| LinearAlgebra/bunchkaufman | X | X | X | X | X |
| LinearAlgebra/svd | X | X | X | X | X |
| LinearAlgebra/lapack | X | X | X | X | X |
| LinearAlgebra/tridiag | X | X | X | X | X |
| LinearAlgebra/bidiag | X | X | X | X | X |
| LinearAlgebra/diagonal | X | X | X | X | X |
| LinearAlgebra/cholesky | X | X | X | X | X |
| LinearAlgebra/lu | X | X | X | X | X |
| LinearAlgebra/symmetric | X | X | X | X | X |
| LinearAlgebra/generic | X | X | X | X | X |
| LinearAlgebra/uniformscaling | X | X | X | X | X |
| LinearAlgebra/lq | X | X | X | X | X |
| LinearAlgebra/hessenberg | X | X | X | X | X |
| LinearAlgebra/blas | X | X | X | X | X |
| LinearAlgebra/adjtrans | X | X | X | X | X |
| LinearAlgebra/pinv | X | X | X | X | X |
| LinearAlgebra/givens | X | X | X | X | X |
| LinearAlgebra/structuredbroadcast | X | X | X | X | X |
| LibGit2/libgit2 | X | X | X | X | X |
| Dates/accessors | X | X | X | X | X |
| Dates/adjusters | X | X | X | X | X |
| Dates/query | X | X | X | X | X |
| Dates/periods | X | X | X | X | X |
| Dates/ranges | X | X | X | X | X |
| Dates/rounding | X | X | X | X | X |
| Dates/types | X | X | X | X | X |
| Dates/io | X | X | X | X | X |
| Dates/arithmetic | X | X | X | X | X |
| Dates/conversions | X | X | X | X | X |
| Base64 | 20 | 0 | 0 | 0 | 1 |
| CRC32c | 658 | 0 | 6 | 0 | 0 |
| DelimitedFiles | X | X | X | X | X |
| FileWatching | X | X | X | X | X |
| Future | 0 | 0 | 0 | 0 | 0 |
| InteractiveUtils | X | X | X | X | X |
| Libdl | X | X | X | X | X |
| Logging | 33 | 1 | 2 | 0 | 0 |
| Markdown | X | X | X | X | X |
| Mmap | X | X | X | X | X |
| Printf | X | X | X | X | X |
| Profile | X | X | X | X | X |
| REPL | 995 | 0 | 0 | 5 | 0 |
| Random | X | X | X | X | X |
| SHA | X | X | X | X | X |
| Serialization | X | X | X | X | X |
| Sockets | X | X | X | X | X |
| Statistics | X | X | X | X | X |
| SuiteSparse | 778 | 0 | 0 | 0 | 0 |
| Test | X | X | X | X | X |
| UUIDs | 22 | 0 | 0 | 0 | 0 |
| Unicode | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |

**Comment by timholy** [comment]
OK, I think this is finally ready to merge. It's failing on nightly because of Debugger's tests; however, presumably Debugger won't pass on nightly without this (specifically, b1f790655cd55c081d29f101a03772ca27714469). I won't merge right away in case @KristofferC wants to use this as a test case for https://github.com/JuliaLang/Pkg.jl/issues/770.

Among many other changes, this now has an interesting "fallback to Compiled" behavior [here](https://github.com/JuliaDebug/JuliaInterpreter.jl/blob/b7028238feaf27fa605f87fe1b7eb937642826c8/test/utils.jl#L181-L191). This makes sure a test runs even if the interpreter gives up on it. Consequently we don't get failures like #30, You can still have problems if a test has done some work (e.g., printing to a buffer) but not finished; when you re-run it under `Compiled` that work will be repeated (which can cause some tests to fail). So this doesn't perfectly eliminate spurious faiilures, but it's getting closer. See 

As a consequence, the tests take *much* longer to run than they used to.

Fixes #36. Fixes #30.

It's now a heck of a lot better, though clearly there's still some ways to go. Here are the new results:

Julia Version 1.2.0-DEV.344
Commit a44b421138* (2019-02-16 16:52 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, haswell)
Environment:
  JULIA_CPU_THREADS = 8
Test run at: 2019-02-17T09:17:51.447

Maximum number of statements per lowered expression: 1000000

| Test file | Passes | Fails | Errors | Broken | Aborted blocks |
| --------- | ------:| -----:| ------:| ------:| --------------:|
| ambiguous | 62 | 0 | 0 | 2 | 0 |
| subarray | 280 | 0 | 1 | 0 | 1 |
| strings/basic | 87290 | 1 | 4 | 0 | 3 |
| strings/search | 522 | 0 | 0 | 0 | 0 |
| strings/util | 449 | 0 | 0 | 0 | 0 |
| strings/io | 12749 | 0 | 0 | 0 | 1 |
| strings/types | 2302685 | 0 | 3 | 0 | 4 |
| unicode/utf8 | 19 | 0 | 0 | 0 | 0 |
| core | X | X | X | X | X |
| worlds | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| keywordargs | 133 | 0 | 1 | 0 | 0 |
| numbers | 1518443 | 27 | 284 | 0 | 6 |
| subtype | X | X | X | X | X |
| char | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| triplequote | 28 | 0 | 0 | 0 | 0 |
| intrinsics | 44 | 0 | 0 | 0 | 0 |
| dict | 144290 | 1 | 1 | 0 | 8 |
| hashing | X | X | X | X | X |
| iobuffer | 200 | 0 | 0 | 0 | 1 |
| staged | 57 | 3 | 0 | 0 | 0 |
| offsetarray | X | X | X | X | X |
| arrayops | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| tuple | 488 | 0 | 1 | 0 | 0 |
| reduce | 8465 | 0 | 3 | 0 | 4 |
| reducedim | X | X | X | X | X |
| abstractarray | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| intfuncs | 4404 | 0 | 15 | 0 | 1 |
| simdloop | X | X | X | X | X |
| vecelement | X | X | X | X | X |
| rational | 97493 | 0 | 29 | 0 | 2 |
| bitarray | 899682 | 0 | 0 | 0 | 9 |
| copy | 493 | 11 | 7 | 0 | 0 |
| math | 1505698 | 0 | 48 | 0 | 5 |
| fastmath | 907 | 3 | 3 | 0 | 0 |
| functional | 95 | 0 | 0 | 0 | 0 |
| iterators | 1568 | 0 | 4 | 0 | 2 |
| operators | 12928 | 0 | 0 | 0 | 1 |
| path | 274 | 0 | 0 | 12 | 2 |
| ccall | X | X | X | X | X |
| parse | 10281 | 20 | 1 | 0 | 1 |
| loading | 143633 | 289 | 1 | 0 | 10 |
| bigint | X | X | X | X | X |
| sorting | X | X | X | X | X |
| spawn | X | X | X | X | X |
| backtrace | 5 | 9 | 12 | 1 | 0 |
| exceptions | 28 | 19 | 6 | 0 | 0 |
| file | X | X | X | X | X |
| read | X | X | X | X | X |
| version | 2468 | 0 | 0 | 0 | 1 |
| namedtuple | 161 | 0 | 8 | 1 | 0 |
| mpfr | 758 | 0 | 45 | 0 | 0 |
| broadcast | 416 | 0 | 7 | 0 | 2 |
| complex | 8249 | 0 | 1 | 2 | 1 |
| floatapprox | 49 | 0 | 0 | 0 | 0 |
| reflection | X | X | X | X | X |
| regex | 38 | 0 | 1 | 0 | 0 |
| float16 | 124 | 0 | 0 | 0 | 0 |
| combinatorics | 95 | 0 | 3 | 0 | 1 |
| sysinfo | 2 | 0 | 0 | 0 | 0 |
| env | X | X | X | X | X |
| rounding | 112496 | 0 | 1 | 0 | 2 |
| ranges | 12108707 | 2 | 209 | 327749 | 7 |
| mod2pi | 80 | 0 | 0 | 0 | 0 |
| euler | 8 | 0 | 4 | 0 | 4 |
| show | X | X | X | X | X |
| client | 0 | 3 | 0 | 0 | 0 |
| errorshow | X | X | X | X | X |
| sets | 739 | 0 | 1 | 0 | 1 |
| goto | X | X | X | X | X |
| llvmcall | X | X | X | X | X |
| llvmcall2 | 6 | 0 | 0 | 0 | 0 |
| grisu | X | X | X | X | X |
| some | 64 | 0 | 0 | 0 | 0 |
| meta | X | X | X | X | X |
| stacktraces | X | X | X | X | X |
| docs | X | X | X | X | X |
| misc | X | X | X | X | X |
| threads | X | X | X | X | X |
| enums | 84 | 3 | 3 | 0 | 0 |
| cmdlineargs | X | X | X | X | X |
| int | 10545 | 0 | 184 | 0 | 1 |
| checked | 1102 | 0 | 9 | 0 | 0 |
| bitset | 192 | 0 | 0 | 0 | 0 |
| floatfuncs | 116 | 0 | 1 | 0 | 1 |
| boundscheck | X | X | X | X | X |
| error | 31 | 0 | 0 | 0 | 0 |
| cartesian | 7 | 0 | 0 | 0 | 0 |
| osutils | 42 | 0 | 0 | 0 | 0 |
| channels | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| iostream | 11 | 0 | 2 | 0 | 0 |
| secretbuffer | 17 | 0 | 0 | 0 | 0 |
| specificity | X | X | X | X | X |
| reinterpretarray | 128 | 0 | 0 | 0 | 1 |
| syntax | X | X | X | X | X |
| logging | 117 | 2 | 0 | 0 | 0 |
| missing | 406 | 0 | 0 | 1 | 1 |
| asyncmap | X | X | X | X | X |
| SparseArrays/higherorderfns | 7031 | 64 | 0 | 73 | 8 |
| SparseArrays/sparse | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| SparseArrays/sparsevector | 9884 | 0 | 1 | 0 | 5 |
| Pkg/resolve | 128 | 1 | 10 | 0 | 1 |
| LinearAlgebra/triangular | 33194 | 0 | 0 | 0 | 2 |
| LinearAlgebra/qr | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| LinearAlgebra/dense | 7692 | 0 | 14 | 0 | 7 |
| LinearAlgebra/matmul | X | X | X | X | X |
| LinearAlgebra/schur | 390 | 0 | 0 | 0 | 1 |
| LinearAlgebra/special | 2863 | 0 | 0 | 0 | 6 |
| LinearAlgebra/eigen | 406 | 0 | 0 | 0 | 2 |
| LinearAlgebra/bunchkaufman | 5145 | 0 | 0 | 0 | 1 |
| LinearAlgebra/svd | X | X | X | X | X |
| LinearAlgebra/lapack | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| LinearAlgebra/tridiag | 1222 | 0 | 0 | 0 | 2 |
| LinearAlgebra/bidiag | 1946 | 0 | 0 | 0 | 1 |
| LinearAlgebra/diagonal | 1619 | 0 | 0 | 0 | 2 |
| LinearAlgebra/cholesky | 2194 | 0 | 0 | 0 | 1 |
| LinearAlgebra/lu | 1182 | 0 | 3 | 0 | 1 |
| LinearAlgebra/symmetric | 2076 | 0 | 0 | 0 | 2 |
| LinearAlgebra/generic | 436 | 0 | 2 | 0 | 3 |
| LinearAlgebra/uniformscaling | X | X | X | X | X |
| LinearAlgebra/lq | 1253 | 0 | 0 | 0 | 1 |
| LinearAlgebra/hessenberg | 40 | 0 | 0 | 0 | 0 |
| LinearAlgebra/blas | 628 | 0 | 0 | 0 | 1 |
| LinearAlgebra/adjtrans | 253 | 0 | 0 | 0 | 1 |
| LinearAlgebra/pinv | 288 | 0 | 0 | 0 | 1 |
| LinearAlgebra/givens | 1840 | 0 | 0 | 0 | 1 |
| LinearAlgebra/structuredbroadcast | 408 | 0 | 0 | 0 | 2 |
| LibGit2/libgit2 | 57 | 0 | 80 | 0 | 0 |
| Dates/accessors | 7723858 | 0 | 0 | 0 | 4 |
| Dates/adjusters | 3147 | 0 | 0 | 0 | 4 |
| Dates/query | 988 | 0 | 0 | 0 | 0 |
| Dates/periods | 679 | 0 | 2 | 0 | 1 |
| Dates/ranges | 349123 | 0 | 0 | 0 | 5 |
| Dates/rounding | 296 | 0 | 0 | 0 | 0 |
| Dates/types | 169 | 0 | 2 | 0 | 0 |
| Dates/io | 100 | 22 | 166 | 0 | 0 |
| Dates/arithmetic | 318 | 0 | 0 | 0 | 0 |
| Dates/conversions | 160 | 0 | 0 | 0 | 0 |
| Base64 | 1015 | 0 | 0 | 0 | 1 |
| CRC32c | 658 | 0 | 6 | 0 | 0 |
| DelimitedFiles | X | X | X | X | X |
| FileWatching | X | X | X | X | X |
| Future | 0 | 0 | 0 | 0 | 0 |
| InteractiveUtils | 96 | 3 | 7 | 0 | 4 |
| Libdl | X | X | X | X | X |
| Logging | 35 | 1 | 0 | 0 | 1 |
| Markdown | X | X | X | X | X |
| Mmap | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |
| Printf | 701 | 38 | 0 | 0 | 0 |
| Profile | 15 | 0 | 0 | 0 | 2 |
| REPL | 995 | 0 | 0 | 5 | 0 |
| Random | X | X | X | X | X |
| SHA | X | X | X | X | X |
| Serialization | X | X | X | X | X |
| Sockets | X | X | X | X | X |
| Statistics | 606 | 0 | 1 | 0 | 4 |
| SuiteSparse | 778 | 0 | 0 | 0 | 0 |
| Test | X | X | X | X | X |
| UUIDs | 22 | 0 | 0 | 0 | 0 |
| Unicode | ☠️ | ☠️ | ☠️ | ☠️ | ☠️ |

**Comment by timholy** [comment]
I will merge this once tests pass.

**Comment by KristofferC** [comment]
Restarted test to pickup Debugger fix

**Comment by timholy** [comment]
There will be an output difference in the Debugger tests, but let's deal with that separately.

**Comment by KristofferC** [comment]
:+1:

**Comment by timholy** [comment]
I've updated the table in #13. Also check out the note about the crashes:

> ones that caused errors or process exit (note: with the possible exceptions of `channels`, `worlds`, and `arrayops`, it appears that most such errors are due to a single cause, #28; deleting [this block](https://github.com/JuliaLang/julia/blob/e5e1253b5ec4ed3fb5b6e12f72dda1dd4348e667/src/ccall.cpp#L1780-L1793) and rebuilding Julia fixes them)

**Pull request merged**